test/parallel/test-tls-multi-key.js

```diff
@@ -2,52 +2,162 @@
 const common = require('../common');
 const assert = require('assert');
 
+// Test multi-identity ('key')/multi-algorithm scenarios.
+
 if (!common.hasCrypto) {
   common.skip('missing crypto');
   return;
 }
 const tls = require('tls');
 const fs = require('fs');
 
-const options = {
+// Key and cert are ordered as ec, rsa.
+test({
   key: [
     fs.readFileSync(common.fixturesDir + '/keys/ec-key.pem'),
     fs.readFileSync(common.fixturesDir + '/keys/agent1-key.pem'),
   ],
   cert: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
     fs.readFileSync(common.fixturesDir + '/keys/agent1-cert.pem'),
-    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem')
-  ]
-};
-
-const ciphers = [];
-
-const server = tls.createServer(options, function(conn) {
-  conn.end('ok');
-}).listen(0, function() {
-  const ecdsa = tls.connect(this.address().port, {
-    ciphers: 'ECDHE-ECDSA-AES256-GCM-SHA384',
-    rejectUnauthorized: false
-  }, function() {
-    ciphers.push(ecdsa.getCipher());
+  ],
+  ca: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ca1-cert.pem'),
+  ],
+});
+
+// Key and cert are in opposite order from each other (order is not required).
+test({
+  key: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-key.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/agent1-key.pem'),
+  ],
+  cert: [
+    fs.readFileSync(common.fixturesDir + '/keys/agent1-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+  ],
+  ca: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ca1-cert.pem'),
+  ],
+});
+
+// Key, cert, and pfx options can be used simultaneously.
+test({
+  key: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-key.pem'),
+  ],
+  cert: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+  ],
+  pfx: fs.readFileSync(common.fixturesDir + '/keys/agent1.pfx'),
+  passphrase: 'sample',
+  ca: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ca1-cert.pem'),
+  ],
+});
+
+// Key and cert with mixed algorithms, and cert chains with intermediate CAs
+test({
```

Comment:
I should have used another proper word. What I worried were that `options.ca` for server you fixed in f6d239e35ddab0950265dad3620593d30aa2b014 which is no use for this test and concatenation intermediate cert of ca4 and ca4 into server certs which results hiding intermediate cert names from the file name. 
This test intends to check two independent cert chains of ECC and RSA works fine so that I think it should be clearly visible what cert chains are testing. Another my proposal is like 
```js
const options = { // [ECC, RSA]
  server: { key: [ec10-key, agent10-key],
            cert: [ec10-cert, agent10-cert],
            ca: [ca6-cert, ca4-cert]},
  client: { ca: [ca5-cert, ca3-cert] }
};
```
with removing `cat ca6-cert.pem >> ec10-cert.pem` and `cat ca4-cert.pem >> agent10-cert.pem` in `Makefile` We can distinguish sever and client options and see what cert chains are going to built in this tls connection.